src/player_display.cpp

```diff
@@ -1104,6 +1080,18 @@ static void draw_tip( const catacurses::window &w_tip, const Character &you,
     wnoutrefresh( w_tip );
 }
 
+inline void skip_skill_headers( const std::vector<HeaderSkill> &skillslist, unsigned int &line,
+                                bool inc, unsigned int line_count )
+{
+    const unsigned int prev_line = line;
+    while( skillslist[line].is_header ) {
```

Comment:
Could this be an endless loop if the only lines are header lines?
Or if line = line_count = 0? - Actually, wouldn't it call for undefined skillslist[0]?

Reply:
Not an endless loop. The conditional break at the bottom of the loop will end it after one cycle.
Good point re line_count == 0, but this is one of many places where the original code has such* a problem and I don't want to try to fix them all in this PR.

\* code paths that could have bugs with arbitrary input but work with every input the game currently produces. In this case, the skillslist is never empty (further, it always starts with a header item, which some other old code takes advantage of and would break if it didn't)